benchmarks/bench_smt_test.go

```diff
@@ -0,0 +1,385 @@
+package smt
```

Comment:
Dude, great job on these tests!

Finding the balance of copy-paste vs abstraction without going overboard either way isn't easy, but you fucking nailed it!